agent/test-agent-cli/design/RUNBOOK.md

```diff
@@ -0,0 +1,118 @@
+
+ENV PKG_CONFIG_PATH=/${ARCH}-bottlerocket-linux-musl/sys-root/usr/lib/pkgconfig
+ENV PKG_CONFIG_ALLOW_CROSS=1
+ENV CARGO_HOME=/src/.cargo
+ENV OPENSSL_STATIC=true
+ENV CARGO_HOME=/src/.cargo
+ADD ./ /src/
+WORKDIR /src/agent/test-agent-cli
+RUN --mount=type=cache,mode=0777,target=/src/target 
+cargo install --locked --offline --target ${ARCH}-bottlerocket-linux-musl --path .  --root ./
+
+FROM public.ecr.aws/amazonlinux/amazonlinux:2
+RUN yum update -y 
+&& yum install -y jq 
+&& yum clean all
+COPY --from=build /src/agent/test-agent-cli/bin/test-agent-cli /usr/local/bin/
+COPY --from=build /src/agent/test-agent-cli/examples/example_test_agent_cli/example-test.sh ./
+
+ENTRYPOINT ["/bin/bash", "./example-test.sh"]
+```
+
+3. Create a docker image consists of the Command Line Interface, Bash Script test and all other resources required to run the test(execute following from Bottlerocket Test System  root folder).(Before running this make sure to add the details to create this image using docker build in [makefile](https://github.com/bottlerocket-os/bottlerocket-test-system/blob/develop/Makefile))
+   `make example-test-agent-cli` and tag this image as : `docker image tag example-test-agent-cli example-test-agent-cli:eks`
+4. Create a controller image(execute from Bottlerocket Test System root folder): `make controller`.
+5. Tag these images as eks as: `docker image tag controller controller:eks`
+6. Create any Resource Agent image(if current test depend on any Resource).
+7. Create a kind cluster that will be used for Testsys `kind create cluster --name testsys`.
+8. Load the docker image for Controller, resource and Test on Kind cluster as :
+```shell
+kind load docker-image controller:eks --name cluster-name
+kind load docker-image example-test-agent-cli:eks --name testsys
+```
+8. Create the namespace, Test and Resource CRDs, roles for test and resource as `testsys install --controller-uri controller:eks`. The controller pod created will run the pod for Test:
+9. Create the resource pod for the Resouce Agent on which the test will be dependent: `testsys run file filename`.
+10. Create a yaml file for Test pod as:
+```shell
+apiVersion: testsys.bottlerocket.aws/v1
+kind: Test
+metadata:
+name: hello-world-cli
+namespace: testsys-bottlerocket-aws
+spec:
+retries: 5
+agent:
+name: hello-agent-cli
+image: example-test-agent-cli:eks
+keepRunning: true
+configuration:
+mode: Fast
+person: Bones the Cat
+helloCount: 3
+helloDurationMilliseconds: 25
+resources: []
+dependsOn: []
+```
+11. Create the Test pod using the image(step 3) and yaml file (step 10) as `testsys run file filename`.
+12. Check the pod status using `kubectl get pods`, the Test pod status will be running if all the dependent resources and Test are available.  This Test pod will run the Bash Test created in step 1, and the output can be seen using `kubectl logs test-pod-name` or `testsys logs test-name` .
```

Comment:
We can do all of this with just testsys so let's use it.

Reply:
This documentation can be less high level because the testsys documentation will go into depth.